stream/ResponseCollector.java

```diff
@@ -61,20 +61,18 @@ synchronized void close(@Nullable StatusRuntimeException error) {
     public static class Queue<R> {
 
         private final BlockingQueue<Either<Response<R>, Done>> responseQueue;
-        private StatusRuntimeException error;
 
         Queue() {
             // TODO: switch LinkedTransferQueue to LinkedBlockingQueue once issue #351 is fixed
             responseQueue = new LinkedTransferQueue<>();
-            error = null;
         }
 
         public R take() {
             try {
                 Either<Response<R>, Done> response = responseQueue.take();
                 if (response.isFirst()) return response.first().message();
-                else if (this.error != null) throw new TypeDBClientException(TRANSACTION_CLOSED_WITH_ERRORS, error);
-                else throw new TypeDBClientException(TRANSACTION_CLOSED);
+                else if (!response.second().error().isPresent()) throw new TypeDBClientException(TRANSACTION_CLOSED);
+                else throw TypeDBClientException.of(response.second().error().get());
```

Comment:
the error is only thrown once, after which the `TRANSACTION_CLOSED` error is thrown